packages/ember-data/lib/serializers/json-serializer.js

```diff
@@ -124,6 +132,101 @@ export default Serializer.extend({
     return data;
   },
 
+  normalizeResponse: function(store, primaryTypeClass, payload, id, requestType) {
+    switch (requestType) {
+      case 'find':
+        return this.normalizeFindResponse(...arguments);
+      case 'findAll':
+        return this.normalizeFindAllResponse(...arguments);
+      case 'findBelongsTo':
+        return this.normalizeFindBelongsToResponse(...arguments);
+      case 'findHasMany':
+        return this.normalizeFindHasManyResponse(...arguments);
+      case 'findMany':
+        return this.normalizeFindManyResponse(...arguments);
+      case 'findQuery':
+        return this.normalizeFindQueryResponse(...arguments);
+      case 'createRecord':
+        return this.normalizeCreateRecordResponse(...arguments);
+      case 'deleteRecord':
+        return this.normalizeDeleteRecordResponse(...arguments);
+      case 'updateRecord':
+        return this.normalizeUpdateRecordResponse(...arguments);
+    }
```

Comment:
could also dry this up to `this[`normalize${capitalize(requestType)}`] || this._normalizeResponse)`